integration-tests/conftest.py

```diff
@@ -24,24 +28,103 @@ def __init__(self, rhsm_register_server_proxy, *args, **kwargs):
         self._private_bus_address = None
 
     def __enter__(self):
-        logger.debug("subscription: starting RHSM private DBus session")
         locale = os.environ.get("LANG", "")
         self._private_bus_address = self._rhsm_register_server_proxy.Start(locale)
-        logger.debug("subscription: RHSM private DBus session has been started")
         return self
 
     def __exit__(self, _exc_type, _exc_value, _exc_traceback):
-        logger.debug("subscription: shutting down the RHSM private DBus session")
         self.disconnect()
         locale = os.environ.get("LANG", "")
         self._rhsm_register_server_proxy.Stop(locale)
-        logger.debug("subscription: RHSM private DBus session has been shutdown")
 
     def _get_connection(self):
         """Get a connection to RHSM private DBus session."""
         # the RHSM private bus address is potentially sensitive
         # so we will not log it
-        logger.info("Connecting to the RHSM private DBus session.")
         return self._provider.get_addressed_bus_connection(
             bus_address=self._private_bus_address, flags=Gio.DBusConnectionFlags.AUTHENTICATION_CLIENT
         )
+
+
+"""
+A class used by a fixture dbus_current_signals.
+A class instance remembers a datetime of a test's start.
+I will provide signals (provided by 'dbus_signals' fixture) that are newer
+than a start of the test.
+"""
+
+
+class DBusSignals:
+    def __init__(self, since, signals):
+        self.since = since
+        self.signals = signals
+
+    """
+    A method for a test to read all signals captured by dbus monitor.
+    The method provides all signals that are emitted after the test begins.
+    """
+
+    def read(self):
+        return [signal for signal in self.signals if signal[0] >= self.since]
+
+
+"""
+A fixture used by tests.
+It is a gate to dbus monitor for a test to get current dbus signals.
+"""
+
+
+@pytest.fixture
+def dbus_current_signals(dbus_signals):
+    return DBusSignals(datetime.now(), dbus_signals)
+
+
+"""
+A fixture that provides a shared object (dbus signals) across processes.
+It is used internally to create an object for dbus signals.
+"""
+
+
+@pytest.fixture(scope="session", autouse=True)
+def dbus_signals():
+    ctx = multiprocessing.get_context("spawn")
+    manager = ctx.Manager()
+    yield manager.list()
+
+
+"""
+A process middleware to run a dasbus loop. The loop requires its own process.
+
+There are subscriptions to DBus events in a callback function.
+
+The function listens on Config/ConfigChanged events
+   - it saves the events into a shared object provided by 'dbus_signals'
+"""
```

Comment:
Same comment here. The doc-string is at wrong position. It should be at line 106.